src/pages/staking/StakeForm.vue

```diff
@@ -273,31 +285,64 @@ export default {
             this.displayConfirmModal = true;
         },
         initiateDeposit() {
+            console.log('initiateDeposit()');
             this.ctaIsLoading = true;
             const value = BigNumber.from(this.topInputAmount);
 
-            this.stlosContractInstance['depositTLOS()']({ value })
-                .then((result) => {
+            let waitTheTransaction: Promise<{hash: string | null}> = Promise.resolve({ hash: null });
+            try {
+                const loginData = localStorage.getItem(LOGIN_DATA_KEY);
+                console.log('initiateDeposit() loginData: ', loginData);
+                if (loginData) {
+                    const loginObj = JSON.parse(loginData);
+                    switch(loginObj?.provider) {
+                    case PROVIDER_TELOS_CLOUD:
+                        waitTheTransaction = this.continueDeposit(value);
+                        break;
+                    default:
+                        waitTheTransaction = this.continueDemositLegacy(value);
+                    }
+                }
+
+                waitTheTransaction.then((result) => {
                     this.resultHash = result.hash;
                     this.$emit('balance-changed');
-                })
-                .catch(({ message }) => {
+                }).catch(({ message }: Error) => {
                     console.error(`Failed to deposit TLOS: ${message}`);
                     this.$q.notify({
                         type: 'negative',
                         message: this.$t('pages.staking.deposit_failed', { message }),
                     });
                     this.resultHash = null;
-                })
-                .finally(() => {
+                }).finally(() => {
                     this.ctaIsLoading = false;
                 });
+
+
+            } catch (e) {
+                console.error('Failed to deposit TLOS', e);
+            } finally {
+                this.ctaIsLoading = false;
+            }
+        },
+        continueDeposit(value: BigNumber) {
+            const authenticator = useAccountStore().getAccount(CURRENT_CONTEXT).authenticator;
+
+            return authenticator.signCustomTransaction(
+                this.stlosContractInstance.address,
+                this.abi,
+                [],
+                value,
+            );
+        },
+        continueDemositLegacy(value: BigNumber) {
```

Comment:
`continueDepositLegacy`

Reply:
In this case, I will argue that we need (at least for now) the two implementations living together. After we can replace that completely with the Antelope lib solution, then we can remove that part.

We can use another name if that's what you mean.

Reply:
OMG!!!!
I just realized what you meant. There was a typo and I argued about the legacy word LOL :rofl: 
Thanks, got it fixed